Wikimate.php

```diff
@@ -374,6 +374,56 @@ public function login($username, $password, $domain = null)
 		return true;
 	}
 
+	/**
+	 * Logs out of the wiki.
+	 *
+	 * @return  boolean  True if logged out
+	 * @link https://www.mediawiki.org/wiki/Special:MyLanguage/API:Logout
+	 */
+	public function logout()
+	{
+		// Obtain logout token first
+		if (($logouttoken = $this->token()) === null) {
+			return false;
+		}
+
+		// Token is needed in MediaWiki v1.34+, older versions produce an
+		// 'Unrecognized parameter' warning which can be ignored
+		$details = array(
+			'action' => 'logout',
+			'token' => $logouttoken,
+		);
+
+		// Send the logout request
+		$response = $this->request($details, array(), true);
+
+		// Check if we got an API result or the API doc page (invalid request)
+		if (strpos($response->body, "This is an auto-generated MediaWiki API documentation page") !== false) {
+			$this->error = array();
+			$this->error['auth'] = 'The API could not understand the logout request';
+			return false;
+		}
+
+		$logoutResult = json_decode($response->body, true);
+		// Check if we got a JSON result
+		if ($logoutResult === null) {
+			$this->error = array();
+			$this->error['auth'] = 'The API did not return the logout response';
+			return false;
+		}
+
+		if ($this->debugMode) {
+			echo "Logout request:\n";
+			print_r($details);
+			echo "Logout response:\n";
+			print_r($logoutResult);
+		}
+
+		// Discard CSRF token for this login session
```

Comment:
Maybe add a note [here](https://github.com/hamstar/Wikimate/blob/3172a7863545f7020cc08fe55192afe83e971c7c/Wikimate.php#L125) and/or [here](https://github.com/hamstar/Wikimate/blob/3172a7863545f7020cc08fe55192afe83e971c7c/Wikimate.php#L242-L243) about when the CSRF token gets discarded?

Reply:
Good idea 1, done.